src/test/java/seedu/address/logic/commands/SortStudentCommandTest.java

```diff
@@ -55,46 +65,148 @@ public void execute_validParameter_success() {
         assertCommandSuccess(command, model, expectedMessage, expectedModel);
     }
 
+    @Test
+    public void execute_sortByName_correctSorting() {
+
+        // add persons in non sorted order
+        model = new ModelManager();
+        model.addPerson(BENSON);
+        model.addPerson(DANIEL);
+        model.addPerson(ALICE);
+        model.addPerson(GEORGE);
+
+        // add persons in sorted order
+        expectedModel = new ModelManager();
+        expectedModel.addPerson(ALICE);
+        expectedModel.addPerson(BENSON);
+        expectedModel.addPerson(DANIEL);
+        expectedModel.addPerson(GEORGE);
+
+        try {
+            // sort by name
+            SortStudentCommand command = new SortStudentCommand(SortStudentParameter.NAME);
+            command.execute(model);
+            ObservableList<Person> expectedPersonList = expectedModel.getAddressBook().getPersonList();
+            ObservableList<Person> actualPersonList = model.getAddressBook().getSortedPersonList();
+            assertEquals(expectedPersonList, actualPersonList);
+        } catch (CommandException e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void execute_sortByEmail_correctSorting() {
+
+        // setup new model for testing correctness of sort
+        Person firstEmail = new PersonBuilder().withEmail("a@example.co").withStudentId("A1234567Z").build();
+        Person secondEmail = new PersonBuilder().withEmail("a@example.com").withStudentId("A1234567F").build();
+        Person thirdEmail = new PersonBuilder().withEmail("aaa@example.com").withStudentId("A1234567A").build();
+        Person fourthEmail = new PersonBuilder().withEmail("b@anotheremail.com").withStudentId("A1234567G").build();
+        Person fifthEmail = new PersonBuilder().withEmail("b@example.com").withStudentId("A1234567L").build();
+
+        // add persons in non sorted order
+        model = new ModelManager();
+        model.addPerson(fifthEmail);
+        model.addPerson(secondEmail);
+        model.addPerson(fourthEmail);
+        model.addPerson(firstEmail);
+        model.addPerson(thirdEmail);
+
+        // add persons in sorted order
+        expectedModel = new ModelManager();
+        expectedModel.addPerson(firstEmail);
+        expectedModel.addPerson(secondEmail);
+        expectedModel.addPerson(thirdEmail);
+        expectedModel.addPerson(fourthEmail);
+        expectedModel.addPerson(fifthEmail);
+
+        try {
+            // sort by email
+            SortStudentCommand command = new SortStudentCommand(SortStudentParameter.EMAIL);
+            command.execute(model);
+            ObservableList<Person> expectedPersonList = expectedModel.getAddressBook().getPersonList();
+            ObservableList<Person> actualPersonList = model.getAddressBook().getSortedPersonList();
+            assertEquals(expectedPersonList, actualPersonList);
+        } catch (CommandException e) {
+            fail();
+        }
+    }
```

Comment:
I would think that since there is a duplicate of code except only when passing in the SortStudentParameter, maybe it will be a good practice to put it into a function, and simply just call the function to reduce the number of lines of code? I notice this "issue" all 3 of the functions you write to test the 3 different parameters.

Reply:
Thanks for the suggestion. I'll refactor the code out into a separate method.